documentation/specs/rar-as-service.md

```diff
@@ -42,14 +42,35 @@ This step will create new process which will act as RAR node. It will also pass
 
 We will use Mutex (as in [Roslyn](https://github.com/dotnet/roslyn/blob/838874b1b817db84ce146bef690cc95a39c213a5/src/Compilers/Server/VBCSCompiler/BuildServerController.cs#L143)) to ensure we don't create two RAR nodes at once. Its name must encode whether it is the user's only RAR node, including user name, administrator privileges, and some initial settings for the node. Such a name could be: `MSBuild.RAR.ostorc.7`, where **MSBuild.RAR** is its prefix, **ostorc** is the user who called MSBuild, and **7** represents encoded settings (flag enum).
```

Comment:
Do we still need a mutex? If multiple nodes try to create a pipe of the same name, all but one will fail. Isn't it enough to implement mutual exclusion?

Reply:
It should be enough -AFAIK -  I wasn't focusing at that area. Will update it.

Reply:
@Ladi I an unaware of 'user's only RAR node' means. Can you please elaborate?

Reply:
This refers to the requirement that there shouldn't be more than one RAR node running for the given user (in a given session?). In other words, we want a singleton process, same as the Roslyn compilation server, for example.

(Looks like you tagged somebody else in your comment.)